src/libraries/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -528,6 +528,56 @@ public SequencePosition GetPosition(long offset)
             return Seek(offset);
         }
 
+        /// <summary>
+        /// Translate opaque <see cref="System.SequencePosition"/> value to numerical position offset
+        /// </summary>
+        /// <param name="position"><see cref="System.SequencePosition"/> to translate to offset</param>
+        /// <returns>Returns the translated offset from the start of the sequence</returns>
+        /// <exception cref="System.ArgumentOutOfRangeException">Thrown when provided <see cref="System.SequencePosition"/> is invalid</exception>
+        public long GetOffset(SequencePosition position)
+        {
+            object? positionSequenceObject = position.GetObject();
+            bool positionIsNotNull = positionSequenceObject != null;
+            BoundsCheck(position, positionIsNotNull);
+
+            object? startObject = _startObject;
+            object? endObject = _endObject;
+
+            uint positionIndex = (uint)GetIndex(position);
+
+            // if sequence object is null we suppose start segment
+            if (!positionIsNotNull)
+            {
+                positionSequenceObject = _startObject;
+                positionIndex = (uint)GetIndex(_startInteger);
+            }
+
+            // Single-Segment Sequence
+            if (startObject == endObject)
+            {
+                return positionIndex;
+            }
+            else
+            {
+                // Multi-Segment Sequence
+                ReadOnlySequenceSegment<T> currentSegment = (ReadOnlySequenceSegment<T>)startObject!;
```

Comment:
Since `currentSegment` can be null, annotate it as nullable.

```suggestion
                ReadOnlySequenceSegment<T>? currentSegment = (ReadOnlySequenceSegment<T>)startObject;
```